stellarphot/settings/custom_widgets.py

```diff
@@ -0,0 +1,391 @@
+# Some settings require custom widgets to be displayed in the GUI. These are defined in
+# this module.
+
+import ipywidgets as ipw
+import traitlets as tr
+from ipyautoui.autoobject import AutoObject
+
+from stellarphot.settings import (
+    Camera,
+    Observatory,
+    PassbandMap,
+    SavedSettings,
+    ui_generator,
+)
+
+
+class ChooseOrMakeNew(ipw.VBox):
+    """
+    Widget to present a list of existing items or the option to make a new one.
+
+    Parameters
+    ----------
+
+    item_type_name : str
+        Name of the item type to be displayed in the widget. Must be one of
+        "camera", "observatory", "passband_map", "Camera", "Observatory"
+        or "PassbandMap".
+    """
```

Comment:
Documentation is missing the keyword `_testing_path`... is that meant to eventually go away?

Reply:
I wasn't sure how to handle this...the intent is that users don't use it, it is just there for testing. I could hide it more, I suppose, by putting it into `**kwargs` and popping it from there. That way it wouldn't show up in code completion.

Reply:
I suppose parameters can't be flagged as private.  But it makes sense that it is for testing purposes.